receiver/memcachedreceiver/integration_test.go

```diff
@@ -57,27 +57,18 @@ func TestIntegration(t *testing.T) {
 	ilms := md.ResourceMetrics().At(0).InstrumentationLibraryMetrics()
 	require.Equal(t, 1, ilms.Len())
 
-	metrics := ilms.At(0).Metrics()
-	require.Equal(t, 5, metrics.Len())
+	aMetricSlice := ilms.At(0).Metrics()
+	require.Equal(t, 11, aMetricSlice.Len())
 
-	assertAllMetricNamesArePresent(t, metadata.Metrics.Names(), metrics)
+	expectedFileBytes, err := ioutil.ReadFile("./testdata/expected_metrics/test_scraper/expected.json")
+	require.NoError(t, err)
 
-	assert.NoError(t, rcvr.Shutdown(context.Background()))
-}
+	unmarshaller := otlp.NewJSONMetricsUnmarshaler()
+	expectedMetrics, err := unmarshaller.UnmarshalMetrics(expectedFileBytes)
+	require.NoError(t, err)
+
+	eMetricSlice := expectedMetrics.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics()
 
-func assertAllMetricNamesArePresent(t *testing.T, names []string, metrics pdata.MetricSlice) {
-	seen := make(map[string]bool, len(names))
-	for i := range names {
-		seen[names[i]] = false
-	}
-
-	for i := 0; i < metrics.Len(); i++ {
-		seen[metrics.At(i).Name()] = true
-	}
-
-	for k, v := range seen {
-		if !v {
-			t.Fatalf("Did not find metric %q", k)
-		}
-	}
+	require.NoError(t, scrapertest.CompareMetricSlices(eMetricSlice, aMetricSlice, false))
+	require.NoError(t, rcvr.Shutdown(context.Background()))
```

Comment:
Why move this? Might as well shut it down as soon as we're done using it

Reply:
Moved to after we collect the metrics.